### DISCUSSION TITLE
Plugin dev and NoReverseMatch

### DISCUSSION THREAD

**Discussion by Barbidul** [netbox-community/netbox#11560] [Q&A] [answered]
Hello. I'm seeking help to understand how the urls registration works. I can't manage to understand why I keep having

****NoReverseMatch at /
'netbox_myplugin' is not a registered namespace inside 'plugins'****

I've made the simplest code with which i get the error : 

setup.py : 
```
from extras.plugins import PluginConfig
class MyPluginConfig(PluginConfig):
    name = "netbox_myplugin"
    verbose_name = "Myplugin"
    description = "Myplugin"
    version = "0.2"
    author = ""
    author_email = ""
    base_url = "myplugin"
    required_settings = []
    default_settings = {}

config = MyPluginConfig
```

urls.py : 

```
from django.urls import path
from django.http import HttpResponse
from . import models, views

def dummy_view(request):
    html = "<html><body>Myplugin</body></html>"
    return HttpResponse(html)

urlpatterns = [
    path("", dummy_view, name="myplugin_list"),
]

```

The plugin appears as installed in the app. When I add a reference to the declared URL, like in navigation.py, I get the error.
Also the URL /plugins/myplugin/ returns a 404.

```
from extras.plugins import PluginMenuItem

menu_items = (
    PluginMenuItem(
        link="plugins:netbox_myplugin:myplugin_list",
        link_text="Myplugin",
    ),
)

```

This happens on a fresh install, version 3.4.3. 
I can install plugins from repositeries and they work. I've spent hours to check for differences with them with no success. Can someone please give me a hint, maybe on how to trace for URLs registration ? 

Thanks a lot

**Comment by Barbidul** [comment, marked as answer]
Solved by touch'ing filtersets.py
The file isn't referenced in my code though.

**Reply by Nikita-Fomichev-DiDi** [reply]
> Решено касанием filtersets.py Однако в моем коде этот файл не упоминается.

Hi, I have the same problem. Tell me, what did you change in the file filtersets.py ?

**Reply by Barbidul** [reply]
I just created an empty file named filtersets.py

**Reply by Nikita-Fomichev-DiDi** [reply]
> I just created an empty file named filtersets.py

Okey, I trying this, but its not helped.
I`m created file filtersets.py in my plugin. And on path netbox/netbox/netbox I see this file filtersets.py is created and is not empty.
It looks like it's something else.
